src/main/java/com/mykaarma/urlshortener/util/UrlServiceUtil.java

```diff
@@ -219,5 +221,35 @@ public String generateValidHash(int hashLength){
 		log.info("Time taken to generate a hash {} is {}ms",shortUrlHash, System.currentTimeMillis()-t1);
 		return shortUrlHash;
 	}
+
+	/**
+	 * This function generates key which will be used by redis for obtain lock.
+	 * Encoding of URL is done to ensure that there are no characters that could potentially interfere with key naming conventions in Redis
+	 * @param businessUuid
+	 * @param longUrl
+	 * @param domainName
+	 * @return lockKey
+	 * @throws Exception
+	 */
+
+	public String getRedisKeyForCreateShortUrl(String businessUuid, String longUrl, String domainName) {
+		String encodedLongUrl = null;
+		String encodedDomainName = null;
+		try {
+			encodedLongUrl = URLEncoder.encode(longUrl, "UTF-8");
+		} catch (Exception e) {
+			log.error(" Could not encode long URL for generating redis key ", e);
+			return null;
+		}
+
+		try {
+			encodedDomainName = URLEncoder.encode(domainName, "UTF-8");
+		} catch (Exception e) {
+			log.error(" Could not encode domain name for generating redis key ", e);
+			return null;
+		}
+		String lockKey = businessUuid + ":" + encodedDomainName + ":" + encodedLongUrl;
```

Comment:
@jatinagarwalmykaarma do we require `encodedDomainName` , won't be businessUuid + encodedLongUrl is sufficient  ?

Reply:
currently we are using these three parameters in mongo db for unique constraint. Hence it makes sense to acquire lock on the combination of these three to avoid conflicts in any future usecase